src/prometheus_async/aio/_decorators.py

```diff
@@ -17,10 +17,23 @@
 """
 
 from time import perf_counter
+from typing import Awaitable, Callable, TypeVar, overload
 
+from typing_extensions import ParamSpec
 from wrapt import decorator
 
 
+P = ParamSpec("P")
+R = TypeVar("R", bound=Awaitable)
+
+
+@overload
+def time(
+    metric,
```

Comment:
How about we define ourselves Protocols for those metrics defining exactly what we need in it? Guess we could import the classes from prometheus_client but i suspect that's potential danger zone.

Reply:
Hm maybe.